frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -1141,9 +1147,14 @@ export class FieldDimension extends Dimension {
     if (this.fk()) {
       const fkDisplayName =
         this.fk() && stripId(this.fk().field().displayName());
-      displayName = `${fkDisplayName} ${FK_SYMBOL} ${displayName}`;
+      if (!displayName.startsWith(`${fkDisplayName} ${FK_SYMBOL}`)) {
```

Comment:
Previously, calling `this.fk().field().displayName()` was pretty bad at actually obtaining a value. Now, it is much better, but sometimes creating a `displayName` using the `fkDisplayName` AND the `displayName` looks awful as the values can repeat. (Example: `Products --> Products --> Category`)